cradlepoint/util/addEntry.js

```diff
@@ -67,4 +68,28 @@ export async function addTestCase(data) {
     } catch (err) {
         throw err
     }
+}
+
+export async function addTestPlan(data) {
+    try {
+        const client = await connectToDb();
+        const valid = await testPlanSchema.isValid(data);
+        console.log(ObjectId.isValid(data.engagementId))
+        if (valid && ObjectId.isValid(data.engagementId)){
```

Comment:
Needs to check that Devices in summaryBOM and that TestCases have valid IDs. I do this in editTestPlan, so you can copy that code.